search/src/main/java/artic/edu/search/DefaultSearchSuggestionsViewModel.kt

```diff
@@ -0,0 +1,165 @@
+package artic.edu.search
+
+import android.support.annotation.DrawableRes
+import com.fuzz.rx.bindTo
+import com.fuzz.rx.disposedBy
+import edu.artic.db.daos.ArticObjectDao
+import edu.artic.db.daos.ArticSearchObjectDao
+import edu.artic.db.models.ArticObject
+import edu.artic.ui.util.asCDNUri
+import edu.artic.viewmodel.BaseViewModel
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.Subject
+import javax.inject.Inject
+
+
+/**
+ * @author Sameer Dhakal (Fuzz)
+ */
+class DefaultSearchSuggestionsViewModel @Inject constructor(searchSuggestionsDao: ArticSearchObjectDao,
+                                                            objectDao: ArticObjectDao
+) : BaseViewModel() {
+
+    private val suggestedKeywords: Subject<List<TextCellViewModel>> = BehaviorSubject.create()
+    private val suggestedArtworks: Subject<List<CircularCellViewModel>> = BehaviorSubject.create()
+    val cells: Subject<List<SearchBaseCellViewModel>> = BehaviorSubject.create()
+
+    private val cellViewHolders: List<SearchBaseCellViewModel>?
```

Comment:
Just use the adapter. we should always have value when these are referenced in the adapter and it needs span size lookup.